parallel-install/pkg/preinstaller/preinstaller_test.go

```diff
@@ -0,0 +1,230 @@
+package preinstaller
+
+import (
+	"fmt"
+	"github.com/avast/retry-go"
+	"github.com/kyma-incubator/hydroform/parallel-install/pkg/config"
+	"github.com/kyma-incubator/hydroform/parallel-install/pkg/logger"
+	"github.com/kyma-incubator/hydroform/parallel-install/pkg/preinstaller/mocks"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/dynamic/fake"
+	"os"
+	"path"
+	"regexp"
+	"testing"
+	"time"
+)
+
+func TestPreInstaller_InstallCRDs(t *testing.T) {
+	scheme := runtime.NewScheme()
+	dynamicClient := fake.NewSimpleDynamicClient(scheme)
+	cfg := getTestingConfig()
+	retryOptions := getTestingRetryOptions(cfg)
+
+	t.Run("should install CRDs", func(t *testing.T) {
+		// given
+		resourceApplierMock := mocks.AllowResourceApplierMock{}
+		i := NewPreInstaller(&resourceApplierMock, cfg, dynamicClient, retryOptions)
+		resourcePath := fmt.Sprintf("%s%s", getTestingResourcesDirectory(), "/correct")
+		input := resourceInfoInput{
+			resourceType:             "CRD",
+			dirSuffix:                "crds",
+			installationResourcePath: resourcePath,
+		}
+
+		// when
+		output, err := i.install(input)
+
+		// then
+		assert.NoError(t, err)
+		assert.True(t, len(output.installed) == 2)
```

Comment:
From my understaning we're not aborting installation on first error, but we continue with remaining resources.
It would be nice to see a test case, where we have: "installed: 3, unInstalled: 1". I don't see such test case?

Reply:
Great point! Added.